src/averageAges.js

```diff
@@ -55,10 +67,41 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const mothers = people.filter((person) => {
+    return people.some((child) => child.mother === person.name);
+  });
+
+  const children = people.filter((child) => {
+    const onlySon = onlyWithSon
+      ? child.sex === 'm'
+      : true;
```

Comment:
as variant you can already immediately return false if this condition does not meets with your requirements
```suggestion
    if (onlyWithSon && child.sex !== 'm') {
      return false;
    }
```